### PR TITLE
onToggle effect

### PR DESCRIPTION
**Pull request by marianozicavo**
Finished the already created `onToggle` effect.

You can toggle a boolean with it, Also, you can directly set it's value. This is useful for cases when you are toggling a boolean in a screen, and after certain user interaction you want to make sure the boolean has a certain value. This is explained more clearly in the documentation.

### CODE REVIEW

**Comment by marianozicavo** [comment]
@ignaciosantise I saw you made the effect, so I assigned you. It would be great if you could give it a look!

File: src/effects/onToggle/docs.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  [actions.TOGGLE]: onToggle()
+};
+
+export default createReducer(initialState, reducerDescription);
+```
+
+If we now do:
+
+```
+dispatch({ type: actions.TOGGLE, target: 'isComplete'});
```
Comment (mvbattan) [inline src/effects/onToggle/docs.md]: Missing space before `}`

File: src/effects/onToggle/docs.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Then the state will be:
+
+```
+state = {
+  isComplete: true
+};
+```
+
+It's also possible to set a custom value directly with this action. This is useful for cases in which you are using the state to determine the visibility of an UI element, like showing an error message that the user can hide. You want to toggle the value from `false` to `true` when the error happens and from `true` to `false` when the user hides the error message. Finally, when the user navigates to a different screen you may want to set the value to `false` to make sure it's not shown on the new screen.
```
Comment (mvbattan) [inline src/effects/onToggle/docs.md]: `different screen, `

File: src/effects/onToggle/docs.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+```
+dispatch({ type: actions.TOGGLE, target: 'isComplete', payload: false});
```
Comment (mvbattan) [inline src/effects/onToggle/docs.md]: Missing space before `}`

File: src/effects/onToggle/docs.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+### Custom selectors
+
+`onToggle` effect receives an optional parameter:
+
+- selector: This function describes how we read the custom value from the `action`.  
+  `(action, state) => any`  
+  By default, is:  
+  `action => action.payload`
```
Comment (mvbattan) [inline src/effects/onToggle/docs.md]: How does this work ?

Comment (mvbattan) [reply src/effects/onToggle/docs.md]: I'd add this to the docs :) Also, I'd use README.md instead of docs.md (check if there is any merge conflicts)

Comment (marianozicavo) [reply src/effects/onToggle/docs.md]: That part is explained in the effect's doc.

Just updated the branch to use README.

**Pull request merged**